### PR TITLE
SFR-2384: Fulfill Manifest Pub Backlist S3 Bucket

### PR DESCRIPTION
**Pull request by mitri-slory**
With this PR I modified the `store_pdf_manifest` method and the prefix attribute of the `FulfillManifestProcess` object to incorporate a new S3 folder called pubBacklist that will include all the manifests from the publisher backlist project such as UofM and UofSC. [https://us-east-1.console.aws.amazon.com/s3/buckets/drb-files-qa?prefix=manifests/&region=us-east-1&bucketType=general](url)

### CODE REVIEW

File: processes/file/fulfill_url_manifest.py [hunk 1]
```diff
@@ -16,15 +16,15 @@ class FulfillURLManifestProcess(CoreProcess):
...
         self.start_timestamp = None
 
         self.generateEngine()
         self.createSession()
 
-        self.s3Bucket = os.environ['FILE_BUCKET']
+        self.s3_bucket = os.environ['FILE_BUCKET']
         self.host = os.environ['DRB_API_HOST']
-        self.prefix = 'manifests/UofM/'
+        self.prefix = 'manifests/pubBacklist/'
```
Comment (kylevillegas93) [inline processes/file/fulfill_url_manifest.py:27]: I recommend actually doing something like publisher_backlist to be explicit with the prefix. @Apophenia what are your thoughts?

File: processes/file/fulfill_url_manifest.py [hunk 2]
```diff
@@ -73,10 +73,8 @@ def update_metadata_object(self, metadata_object, bucket_name, curr_key):
             metadata_json, counter = self.reading_order_fulfill(metadata_json, counter)
             metadata_json, counter = self.resource_fulfill(metadata_json, counter)
             metadata_json, counter = self.toc_fulfill(metadata_json, counter)
-        except (Exception, IndexError) as e:
-            logger.error(e)
-        except:
-            logger.error('One of the Link fulfill methods failed')  
+        except Exception as e:
```
Comment (kylevillegas93) [inline processes/file/fulfill_url_manifest.py:76]: Nice!

File: services/sources/publisher_backlist_service.py [hunk 3]
```diff
@@ -133,9 +133,9 @@ def store_pdf_manifest(self, record):
 
             if media_type == 'application/pdf':
                 record_id = record.identifiers[0].split('|')[0]
-                manifest_path = 'manifests/{}/{}.json'.format(source, record_id)
+                manifest_path = 'manifests/pubBacklist/{}/{}.json'.format(source, record_id)
```
Comment (kylevillegas93) [inline services/sources/publisher_backlist_service.py:136]: Could you use self.prefix here so that you don't have to update this prefix in two places?

Comment (mitri-slory) [reply services/sources/publisher_backlist_service.py:136]: Sure I can do that

**Review by kylevillegas93 — APPROVED** [review]

File: processes/file/fulfill_url_manifest.py (hunk 2 above)
Comment (Apophenia) [reply processes/file/fulfill_url_manifest.py:76]: Yes, I think this looks great!

File: services/sources/publisher_backlist_service.py [hunk 4]
```diff
@@ -133,9 +134,9 @@ def store_pdf_manifest(self, record):
 
             if media_type == 'application/pdf':
                 record_id = record.identifiers[0].split('|')[0]
-                manifest_path = 'manifests/{}/{}.json'.format(source, record_id)
+                manifest_path = f'{self.prefix}/{source}/{record_id}.json'
```
Comment (Apophenia) [inline services/sources/publisher_backlist_service.py:137]: I think this looks good!

**Pull request merged**